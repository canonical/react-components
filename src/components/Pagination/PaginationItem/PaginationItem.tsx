import classNames from "classnames";
import React from "react";
import type { MouseEventHandler } from "react";

export type Props = {
  /**
   * Whether the pagination item is active, i.e. the current page is this page.
   */
  isActive?: boolean;
  /**
   * The page number.
   */
  number: number;
  /**
   * Function to handle clicking the pagination item.
   */
  onClick: MouseEventHandler<HTMLButtonElement>;
};

const PaginationItem = ({
  number,
  onClick,
  isActive = false,
}: Props): JSX.Element => (
  <li className="p-pagination__item">
    <button
      className={classNames("p-pagination__link", {
        "is-active": isActive,
      })}
      onClick={onClick}
      aria-current={isActive ? "page" : undefined}
    >
      {number}
    </button>
  </li>
);

export default PaginationItem;
