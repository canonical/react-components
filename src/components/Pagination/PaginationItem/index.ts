export { default } from "./PaginationItem";
export type { Props as PaginationItemProps } from "./PaginationItem";
