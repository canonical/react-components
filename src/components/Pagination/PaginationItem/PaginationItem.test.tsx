import React from "react";
import { render, screen } from "@testing-library/react";

import PaginationItem from "./PaginationItem";

describe("<PaginationItem />", () => {
  // snapshot tests
  it("renders and matches the snapshot", () => {
    const { container } = render(
      <PaginationItem number={1} onClick={jest.fn()} />
    );

    expect(container).toMatchSnapshot();
  });

  // unit tests
  it("displays the page number", () => {
    render(<PaginationItem number={1} onClick={jest.fn()} isActive />);

    expect(screen.getByRole("button", { name: "1" })).toBeInTheDocument();
  });

  it("sets isActive", () => {
    render(<PaginationItem number={1} onClick={jest.fn()} isActive />);

    expect(screen.getByRole("button")).toHaveClass("is-active");
  });

  it("sets aria-current when isActive is true", () => {
    render(<PaginationItem number={1} onClick={jest.fn()} isActive />);

    expect(screen.getByRole("button", { current: "page" })).toBeInTheDocument();
  });
});
