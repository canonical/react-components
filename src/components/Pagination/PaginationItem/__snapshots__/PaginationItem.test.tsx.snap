// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationItem /> renders and matches the snapshot 1`] = `
<div>
  <li
    class="p-pagination__item"
  >
    <button
      class="p-pagination__link"
    >
      1
    </button>
  </li>
</div>
`;
